MetaMask/metamask-extension
# Github action to check issue body matches issues templates and add labels to issue when needed

## **Description**

This PR introduces a new Github action, to achieve the following:
1. Check if issue matches an existing issue template ('general-issue.yml' or 'bug-report.yml')
2. Add an `INVALID-ISSUE-TEMPLATE` label in case issue doesn't match any of the templates
3. Add an `external-contributor` label in case issue was created by someone external to the MetaMask organisation
4. Add a `regression-prod-x.y.z` label when possible, when `x.y.z` value can be extracted from 'bug-report.yml' template

We will be able to leverage `regression-prod-x.y.z` and `external-contributor` labels to improve our bug triage process and our metrics collection.

[Same PR for Mobile repo](https://github.com/MetaMask/metamask-mobile/pull/7379)

## **Manual testing steps**

1. Go on this [test repo](https://github.com/gauthierpetetin-test/repo_test/issues/new?assignees=&labels=type-bug&projects=&template=bug-report-extension.yml&title=%5BBug%5D%3A+)
2. Fill the bug report issue form
6. Make sure the release version you indicate in the form has x.y.z format
7. Click on "Submit new issue"
8. Wait for ~30s and 2 new labels shall appear on the issue you just created (`regression-prod-x.y.z` and `external-contributor`)
9. Edit the issue's description by indicating a different release version
10. Wait for ~30s and the `regression-prod-x.y.z` shall be updated accordingly
11. Edit the issue by deleting the whole description
12. Wait for ~30s and the `INVALID-ISSUE-TEMPLATE` shall be added
 
## **Screenshots/Recordings**

### **Before**

NA

### **After**

https://recordit.co/Sq7fvCwkr5

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1409

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
